src/Bicep.Core/TypeSystem/ObjectType.cs

```diff
@@ -37,6 +38,9 @@ public ObjectType(string name, TypeSymbolValidationFlags validationFlags, IEnume
 
         public TypePropertyFlags AdditionalPropertiesFlags { get; }
 
+        public bool HasAdditionalPropertiesType =>
+            !(AdditionalPropertiesType == null || AdditionalPropertiesFlags.HasFlag(TypePropertyFlags.FallbackProperty));
```

Comment:
[nit]: From the property name, I would expect `<ObjectType>.HasAdditionalPropertiesType` to be synonymous with `<ObjectType>.AdditionalPropertiesType != null`. Maybe it should be called `HasExplicitAdditionalPropertiesType` instead?